addons/hr/models/res_partner.py

```diff
@@ -22,3 +22,17 @@ def get_static_mention_suggestions(self):
             return suggestions
         except AccessError:
             return suggestions
+
+    @api.multi
+    def name_get(self):
+        """ Override to allow an employee to see its private address in his profile.
+            This avoids to relax access rules on `res.parter` and to add an `ir.rule`.
+            (advantage in both security and performance).
+            Use a try/except instead of systematically checking to minimize the impact on performance.
+            """
+        try:
+            return super(Partner, self).name_get()
+        except AccessError as e:
+            if self.ensure_one() and self.env.user.employee_ids.mapped('address_home_id') == self:
```

Comment:
if len(self) == 1 and self in self.env.user.employee_ids.mapped('address_home_id'):

Avoid another error in the Accesserror catch ;) and an user can have several employees.